File: dotnet/DesignPatterns/StructuralDesignPatterns/CompositeDesignPattern/CompositeDesignPattern.Example3/Manager.cs
namespace CompositeDesignPattern.Example3;

public class Manager : EmployeeComposite
{
	public Manager(string name, double salary) : base(name, salary)
	{
	}

	public override void Print()
	{
		Console.WriteLine($"Manager Name\t\t: {GetName()}");
		Console.WriteLine($"Manager Salary\t\t: {GetSalary()}");
		double totalSalary = 0;
		_employees.ForEach(employee =>
		{
			totalSalary += employee.GetSalary();
			employee.Print();
		});
		Console.WriteLine($"Total Salary\t: {totalSalary}");
	}
}

File: nodejs/DesignPatterns/StructuralDesignPatterns/CompositeDesignPattern/CompositeDesignPattern.Example3/src/Manager.ts
import { EmployeeComposite } from './EmployeeComposite';

export class Manager extends EmployeeComposite {
  constructor(name: string, salary: number) { super(name, salary); }

  print(): void {
    console.log(`Manager Name\t\t: ${this.getName()}`);
    console.log(`Manager Salary\t\t: ${this.getSalary()}`);
    let totalSalary = 0;
    for (const e of this.employees) {
      totalSalary += e.getSalary();
      e.print();
    }
    console.log(`Total Salary\t: ${totalSalary}`);
  }
}
